### PR TITLE
Outline Hubot Evolution process

### PR DESCRIPTION
**Pull request by bkeepers**
Here is a process largely lifted from [Swift Evolution](https://github.com/apple/swift-evolution) for proposing and discussing new features for Hubot. Check out the [README](https://github.com/hubotio/evolution/blob/master/README.md) for an overview.

The largest change from Swift Evolution is making the process entirely based on Pull Requests, instead of a hybrid of mailing list+Pull Requests+website.

cc @hubotio/maintainers @technicalpickles @mose @gr2m

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+1. **Socialize the idea**: Before starting the review process, it's helpful to gauge interest from the community. Ideally, the idea would have come up in a discussion in another issue or in [chat][] and the consensus was to start an evolution proposal.
+1. **Develop the proposal**: open a Pull Request with a rough sketch of the proposal using the [proposal template](template.md) in the [_drafts](https://github.com/hubotio/evolution/new/master/_drafts) directory. Prototyping an implementation and its uses along with the proposal is encouraged, because it helps ensure both technical feasibility of the proposal as well as validating that the proposal solves the problems it is meant to solve.
+1. **Request a review**: When you are ready for feedback from the Hubot community, change the frontmatter in your proposal from `status: in-progress` to `status: in-review`.
```
Comment (bkeepers) [inline CONTRIBUTING.md]: Anyone have a better idea for how to manage this step? We need a way to announce to the community that a proposal is ready for review.

I was thinking we would add some automation here where changes to the frontmatter of the proposal would update the labels on the PR, cc @hubotio/maintainers, and post an announcement in Slack.

Comment (gr2m) [reply CONTRIBUTING.md]: I think changing the label is good enough to start. I would reflect on the process as we go and see what we can improve. We will hopefully have people who made proposals already and went trough the process, getting their feedback will help us prioritize on the parts of the process we should improve.

**Review by gr2m — APPROVED** [review]
I think this is a great start that we can iterate on as we go

**Comment by bkeepers** [comment]
I created a [project board](https://github.com/hubotio/evolution/projects/1) and installed a [state](https://github.com/probot/state) integration that will automatically update the board as lablels are added and proposals move through the process.

I think this is good to go.

**Review by mose — APPROVED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+Once a proposal has been implemented, the Pull Request will be merged.
+
+## Proposal states
+
+A proposal can be in one of several states:
+
+* **In Progress**: The authors are still working on the proposal and are not yet ready for review.
+* **In Review**: The proposal is ready for review and comments from the community.
+* **Withdrawn**: The proposal has been withdrawn by the original submitter.
+* **Deferred**: Consideration of the proposal has been deferred because it does not meet the [goals of the upcoming release](README.md). Deferred proposals will be reconsidered when scoping the next major release.
```
Comment (technicalpickles) [inline CONTRIBUTING.md:53]: I really ❤️ this state. I think there has always been some tension when adding new features, where I haven't wanted to say no to them, but now just didn't feel like the right time to do them. Deferred feels right in this context, as it doesn't mean no, just not now, with a concrete way to revisit them after the next major release.

**Review by technicalpickles — APPROVED** [review]
This looks great ✨ 

At first glance, I was a little apprehensive about how many states there were, and if maybe this was a bit too much process. Reading over the rest though, I think it's laid out really well, so I'm not as concerned with that.

**Comment by bkeepers** [comment]
Thanks for the review, everyone! Now that this has been accepted, the last step is to add it to the contribution docs. I'm going to go ahead an merge this so I can start linking to it, and will update the labels once it's fully implemented.

**Pull request merged**